### PR TITLE
support waypoints

### PR DESCRIPTION
**Pull request by YusukeHata**
support waypoints.

Would you read sample code as follow?

```java
Routing routing = new Routing(Routing.TravelMode.DRIVING);
routing.registerListener(this);
routing.execute(start, waypoints1, waypoints2, end);
```

### CODE REVIEW

File: library/src/main/java/com/directions/route/Routing.java [hunk 1]
```diff
@@ -13,8 +16,17 @@ public Routing(TravelMode mTravelMode) {
     }
 
     protected String constructURL(LatLng... points) {
-        LatLng start = points[0];
-        LatLng dest = points[1];
+        if (points.length <= 1) {
```
Comment (deepankarb) [inline library/src/main/java/com/directions/route/Routing.java]: You sure you can route with a single point?

File: library/src/main/java/com/directions/route/Routing.java [hunk 2]
```diff
@@ -13,8 +16,17 @@ public Routing(TravelMode mTravelMode) {
...
+            throw new IllegalArgumentException("points length : " + points.length);
```
Comment (deepankarb) [inline library/src/main/java/com/directions/route/Routing.java]: What motivation for plagiarism on a collaborative site?

File: library/src/main/java/com/directions/route/Routing.java [hunk 3]
```diff
@@ -27,6 +39,15 @@ protected String constructURL(LatLng... points) {
         mBuf.append(dest.longitude);
         mBuf.append("&sensor=true&mode=");
         mBuf.append(_mTravelMode.getValue());
+        if (waypoints.size() > 0) {
```
Comment (deepankarb) [inline library/src/main/java/com/directions/route/Routing.java]: What motivation for plagiarism on a collaborative site?

Comment (YusukeHata) [inline library/src/main/java/com/directions/route/Routing.java]: I have just knew your waypoint pull request.
I am sorry if I offended you.
I don't plagiarize.
I'm not good at English.sorry.

File: library/src/main/java/com/directions/route/Routing.java (hunk 1 above)
Comment (YusukeHata) [inline library/src/main/java/com/directions/route/Routing.java]: its bug. thank you.

**Comment by jd-alexander** [comment]
Already added support for this from @deepankarb

**Pull request closed without merging**